kv/txn_coord_sender.go

```diff
@@ -307,15 +307,22 @@ func (tc *TxnCoordSender) Send(ctx context.Context, call proto.Call) {
 	ctx = tracer.ToCtx(ctx, trace)
 
 	// Process batch specially; otherwise, send via wrapped sender.
-	switch args := call.Args.(type) {
+	switch batchArgs := call.Args.(type) {
```

Comment:
What I'm doing at the replica level is checking whether or not something is a BatchRequest first thing; if it's not, then I wrap it in a batch. Then _everywhere_ downstream, I only ever deal with batches. It simplifies things.

Here, it might make sense to change proto.Call to just take a batch. Is that planned for a later PR?